src/lib/plugin.js

```diff
@@ -157,21 +157,49 @@ module.exports = class PluginVirtual extends EventEmitter2 {
   }
 
   async sendRequest (message) {
+    this._validator.validateOutgoingMessage(message)
+    this._safeEmit('outgoing_request', message)
+
     // for backward compat with ilp-kit version < 2.1.0, copy `.custom` to `.data`.
     // This will only work in connector-to-connector send_message rpc calls as implemented here in
     // ilp-plugin-virtual. When sent over a FiveBellsLedger, like in five-bells-integration-test,
     // message.data will be stripped again by ilp-plugin-bells, because it does not comply with
     // the Message schema as defined by five-bells-shared.
-    message.data = message.custom
-
-    this._validator.validateOutgoingMessage(message)
-    this._safeEmit('outgoing_request', message)
+    // Using JSON stringify-then-parse here as an object cloning method, since Object.assign only
+    // makes a shallow copy, and message.custom is expected to have nested objects in it.
+    const convertedMessage = JSON.parse(JSON.stringify(message))
+    convertedMessage.data = convertedMessage.custom
+    delete convertedMessage.custom
```

Comment:
Why is this being deleted? the custom field is what should be used, and the `data` field should just be kept around for backwards compatibility